archlinuxcn/3dslicer-bin/lilac.py

```diff
@@ -5,15 +5,16 @@
 
 def pre_build():
   update_pkgver_and_pkgrel(_G.newver.lstrip('v'))
-  cmd = "curl https://download.slicer.org/ |grep bitstream |cut -d '/' -f 3 |cut -d '\"' -f 1 |sed -n 3p"
-  newstr = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8')[0:-1]
+  cmd = "curl https://download.slicer.org/ | grep bitstream | cut -d '\"' -f 2"
+  newstr = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8')
+  newstr = newstr.split('\n')[2].replace('/bitstream/', '').replace('/bitstream/', '')
 
-  cmd = "grep bitstream PKGBUILD|cut -f 5 -d '/'
+  cmd = "grep bitstream PKGBUILD|cut -f 5 -d \'/\'"
   oldstr = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8')[0:-2]
 
   if oldstr == newstr:
     raise RuntimeError("URL is not updated yet.")
-  cmd = f"sed -i '/bitstream/s/{oldstr}/{newstr}/' PKGBUILD"
+  cmd = f"sed -i \'/bitstream/s/{oldstr}/{newstr}/\' PKGBUILD"
```

Comment:
This doesn't solve the issue when `oldstr` or `newstr` contains unexpected content.

Please, avoid using `shell=True` when there is input from outside.